xdsl/dialects/csl/csl_wrapper.py

```diff
@@ -142,12 +143,18 @@ class ModuleOp(IRDLOperation):
 
     The layout module has two additional block args `x` and `y` as part of the `@set_tile_code` loop nest.
     Operations using these args need to be lowered to the correct place in the loop nest.
+
+    The program module has the following args (in order):
+      * general params:        `width` and `height` followed by everything specified in `params`
+      * params from layout:    everything defined by `layout_yield_op.fields`
+      * input-output symbols:  any arg from the function lowered into this op, which are exported symbols supporting host-device transfers
```

Comment:
Are there some properties of these in-out symbols that we need to verify?

Reply:
Can they be anything, or do we have some type restrictions? I'm asking because I'm not sure myself. I would guess that they need to be `memref`?

Reply:
@dk949 ?

Reply:
Actually, I noticed we're using it for stencil.fields right now. My suggestion is to not restrict it for the time being.